src/hotspot/share/oops/cpCache.cpp

```diff
@@ -690,15 +690,15 @@ void ConstantPoolCache::save_for_archive() {
     copy[i] = *entry_at(i);
   }
 
-  SystemDictionaryShared::save_cpcache_entries(constant_pool()->pool_holder(), copy);
+  SystemDictionaryShared::set_saved_cpcache_entries(this, copy);
 #endif
 }
 
 void ConstantPoolCache::remove_unshareable_info() {
 #if INCLUDE_CDS
   Arguments::assert_is_dumping_archive();
-  InstanceKlass* ik = constant_pool()->pool_holder();
-  ConstantPoolCacheEntry* saved = SystemDictionaryShared::get_saved_cpcache_entries_locked(ik);
+  ConstantPoolCache* orig_cpc = ArchiveBuilder::current()->get_src_obj(this);
+  ConstantPoolCacheEntry* saved = SystemDictionaryShared::get_saved_cpcache_entries_locked(orig_cpc);
```

Comment:
What does this do?  get_src_obj looks like it just returns the address with a cast to the type T.

Reply:
I was confused with the template.  get_src_obj() gets in-memory (current) version of the metadata object.  It has a mapping to the copy in the archive. right?
"this" is the copy in the archive.  The comment helps.